src/layers/core/hexagon-layer/hexagon-layer.js

```diff
@@ -120,25 +121,39 @@ export default class HexagonLayer extends Layer {
   }
 
   updateRadiusAngle() {
-    const {hexagonVertices: vertices, dotRadius} = this.props;
+    let vertices = this.props.hexagonVertices;
+    let angle;
+    let radius;
 
-    const vertex0 = vertices[0];
-    const vertex3 = vertices[3];
+    if (!Array.isArray(vertices) || vertices.length !== 6) {
+      vertices = _getDefaultVertices(this.props.data, this.props.getCentroid);
+    }
 
-    // transform to space coordinates
-    const spaceCoord0 = this.projectFlat(vertex0);
-    const spaceCoord3 = this.projectFlat(vertex3);
+    if (!vertices) {
+      angle = 0;
+      radius = 10;
+    } else {
 
-    // distance between two close centroids
-    const dx = spaceCoord0[0] - spaceCoord3[0];
-    const dy = spaceCoord0[1] - spaceCoord3[1];
-    const dxy = Math.sqrt(dx * dx + dy * dy);
+      const vertex0 = vertices[0];
+      const vertex3 = vertices[3];
+
+      // transform to space coordinates
+      const spaceCoord0 = this.projectFlat(vertex0);
+      const spaceCoord3 = this.projectFlat(vertex3);
+
+      // distance between two close centroids
+      const dx = spaceCoord0[0] - spaceCoord3[0];
+      const dy = spaceCoord0[1] - spaceCoord3[1];
+      const dxy = Math.sqrt(dx * dx + dy * dy);
 
-    this.setUniforms({
       // Calculate angle that the perpendicular hexagon vertex axis is tilted
-      angle: Math.acos(dx / dxy) * -Math.sign(dy) + Math.PI / 2,
-      // Allow user to fine tune radius
-      radius: dxy / 2 * Math.max(0, Math.min(1, dotRadius))
+      angle = Math.acos(dx / dxy) * -Math.sign(dy) + Math.PI / 2;
+      radius = dxy / 2;
+    }
+
+    this.setUniforms({
```

Comment:
My assumption was to just generate a list of hexagon vertices from angle and radius (if hexagonVertices was not provided). In this version, we would not need angle and radius as uniforms.

This function would be called `calculateDefaultHexagonVertices` or similar.